pyalarmdotcomajax/pyalarmdotcomajax.py

```diff
@@ -205,31 +223,46 @@ async def async_login(self):
             return False
         return await self._async_get_system_info()
 
-    async def async_update(self):
-        """Fetch the latest state."""
+    async def async_update(self, mode="alarm"):
+        """Fetch the latest state according to mode."""
         _LOGGER.debug("Calling update on Alarm.com")
         if not self._ajax_headers["ajaxrequestuniquekey"]:
             await self.async_login()
         try:
-            # grab partition status
-            async with self._websession.get(
-                url=self.PARTITION_URL_TEMPLATE.format(
+            if mode == "lock":
+                state_url = self.LOCK_URL_TEMPLATE.format(self._url_base, self._lockid)
+            else:
+                state_url = self.PARTITION_URL_TEMPLATE.format(
                     self._url_base, self._partitionid
-                ),
+                )
+
+            # grab status
+            async with self._websession.get(
+                url=state_url,
                 headers=self._ajax_headers,
             ) as resp:
                 json = await (resp.json())
-            self.sensor_status = json["data"]["attributes"]["needsClearIssuesPrompt"]
-            self.sensor_status = (
-                "System needs to be cleared" if self.sensor_status else "System OK"
-            )
             self.state = json["data"]["attributes"]["state"]
-            self.state = self.STATEMAP[self.state]
+            if mode == "lock":

```

Comment:
Looks like the `if else` here and the `if else` below are just the same two branches - let's combine them

Reply:
Not sure about that. I am thinking of the first if as something that can be extended to support other things alarm.com, but the "needs clearing" firmly belongs to the alarm only.

Reply:
So you are saying there may be more modes in the future so they might be different. But as there are currently only 2 modes, these are the same for now right? We can combine them for now and split them later if necessary.

Reply:
Done.